tests/src/common/lease.rs

```diff
@@ -215,17 +215,24 @@ pub(crate) fn complete_initialization<Lpn, DownpaymentC, LeaseC>(
         exp_borrow,
     );
 
-    do_swap(app, lease_addr, connection_id, downpayment, exp_lease);
+    do_swap(
+        app,
+        lease_addr,
+        ica_addr,
+        connection_id,
+        (downpayment, exp_borrow, exp_lease),
+    );
 }
 
-fn do_swap<DownpaymentC, LeaseC>(
+fn do_swap<DownpaymentC, Lpn, LeaseC>(
```

Comment:
why do we organize the requests and responses from the perspective of the contract? That turns into mixing responses for a previous step with requests for the next step. For example, the blank response from DEX should be at the end of the previous step, the transfer-out, not the first step of the swap.